src/main/io/gps.c

```diff
@@ -878,30 +1364,40 @@ void gpsUpdate(timeUs_t currentTimeUs)
 
         case GPS_STATE_LOST_COMMUNICATION:
             gpsData.timeouts++;
-            if (gpsConfig()->autoBaud) {
+            // commented out, as it is dangerous and should not be used, fix for a correct auto baud in another pr
+            /*if (gpsConfig()->autoBaud) {
                 // try another rate
                 gpsData.baudrateIndex++;
                 gpsData.baudrateIndex %= GPS_INIT_ENTRIES;
-            }
+            }*/
             gpsSol.numSat = 0;
             DISABLE_STATE(GPS_FIX);
             gpsSetState(GPS_STATE_INITIALIZING);
             break;
 
         case GPS_STATE_RECEIVING_DATA:
             // check for no data/gps timeout/cable disconnection etc
-            if (millis() - gpsData.lastMessage > GPS_TIMEOUT) {
+            DEBUG_SET(DEBUG_GPS_UNIT_CONNECTION, 2, millis() - gpsData.lastMessage);
```

Comment:
going through the code again, I strongly suggest splitting this into at least two (preferably 3), unrelated PRs (the current one is huge and hard to reason about):

- Adding Debug info about GPS
- Adding some extra reconnect mechanisms to improve connection stability (?)
- Adding M10 protocol support

While all of these are important, I just don't see how these belong in the same PR

Reply:
This PR also adds some other things as well:
- configurable UTC standard (why do we need this?)
- adds AssistNow via NAV5X (useful, but to your point makes the PR bigger)